### PR TITLE
Feat/value transform

### PR DESCRIPTION
**Pull request by ArrayKnight**
Built on top of https://github.com/selbekk/calidation/pull/43

### CODE REVIEW

**Comment by ArrayKnight** [comment]
@selbekk Check out PR https://github.com/selbekk/calidation/pull/43 first. This one builds on top of that. But is an unrelated feature. (My assumption is https://github.com/selbekk/calidation/pull/43 should go through relatively painlessly, but this one might require a bit more discussion).

This PR adds the ability to modify the value before it's stored into state and then validated. In my particular use case, I have a desire to have `fields` represent my model with correct data types (arrays, booleans, numbers, strings, etc). This means I need to transform the value from a string into X, and then back to a string (sometimes) when setting `value`. It's sort of round about, but it's necessary for certain components where I don't have control over the inner workings. It also means that `onSubmit` needs to extra conversion, it can simply pass `fields`.

File: README.md [hunk 1]
```diff
@@ -520,6 +537,25 @@ Each validator can accept an object with a `message` key or - in the case where
 you don't have to specify anything other than a validation message - just a
 string with the error message.
 
+##### `initialValues: object`
+
+The `initialValues` object lets you specify the initial values of the form
+fields. These values are available from the `fields` argument in the `children`
+function, which lets you control your form fields.
```
Comment (selbekk) [inline README.md]: Can you remove the same text from line 432-436?

File: src/Form.js [hunk 2]
```diff
@@ -33,19 +33,24 @@ class Form extends Component {
...
+            typeof this.transforms[name] === 'function'
+                ? this.transforms[name]
+                : value => value;
+
         this.setField({
-            [e.target.name]:
-                e.target.type === 'checkbox'
-                    ? e.target.checked
-                    : e.target.value,
+            [name]: transform(type === 'checkbox' ? checked : value),
```
Comment (selbekk) [inline src/Form.js]: A stylistic thing, but it also avoids creating an arrow function when not required.

How about?

```js
let finalValue = type === 'checkbox' ? checked : value;
if (transform[e.target.name]) {
  finalValue = transform[e.target.name](value);
}
this.setField({ [name]: finalValue });
```

File: src/Validation.js [hunk 3]
```diff
@@ -31,31 +34,44 @@ class Validation extends Component {
...
+        const { config, initialValues, register, valueTransforms } = this.props;
+
+        register(
+            config,
+            valueTransforms,
+            Object.keys(config).reduce((allFields, field) => {
+                const transform =
+                    typeof valueTransforms[field] === 'function'
+                        ? valueTransforms[field]
+                        : value => value;
```
Comment (selbekk) [inline src/Validation.js]: Same as above

File: README.md (hunk 1 above)
Comment (ArrayKnight) [reply README.md]: I removed all of the duplication and added a note to `FormValidation` to look at `Form` & `Validation` for the props. Let me know if this is acceptable.

**Comment by ArrayKnight** [comment]
I'm glad you didn't like `valueTransforms`. I initially went with `transforms` but updated for additional clarity trying to anticipate your feedback.

File: README.md (hunk 1 above)
Comment (selbekk) [reply README.md]: Yeah that's fine.

**Comment by selbekk** [comment]
Good stuff buddy!

**Pull request merged**